### ISSUE TITLE
Share extensions crash on iOS 10

### ISSUE THREAD

**Issue by rob-keepsafe** [bugsnag/bugsnag-cocoa#411]
### Description
Use Bugsnag within a share extension and support iOS 10

The podfile specifies [iOS 8+ is supported](https://github.com/bugsnag/bugsnag-cocoa/blob/master/Bugsnag.podspec.json#L19)

### Issue
We experienced a lot of KVO crashes in production

<img width="638" alt="Screen Shot 2019-09-09 at 2 31 56 PM" src="https://user-images.githubusercontent.com/30269720/64987624-93a16500-d897-11e9-8569-7731ae69f1cf.png">

<img width="1371" alt="Screen Shot 2019-09-16 at 3 36 58 PM" src="https://user-images.githubusercontent.com/30269720/64988144-cdbf3680-d898-11e9-9a8b-32d9959fe1ad.png">

### Environment

Library versions:

* Bugsnag version (from your Podfile, Podfile.lock, or elsewhere): 5.22.2
* CocoaPods version (if any) (`pod -v`): 1.7.5
* iOS/tvOS/macOS version(s) 10
* debug mode or production?: production
* How are you initializing Bugsnag? Can you share a snippet of your
  `AppDelegate` where Bugsnag is configured?

```
    // Apple doc suggests grabbing extensionContext from loadView
    override func loadView() {
        super.loadView()
        guard let extensionContext = self.extensionContext else { return }
        // TODO: this is crashing the extension for KVO dealloc errors
//        Bugsnag.start(withApiKey: "....")
    }
```

### Stack trace

```
Hardware Model:     iPhone5,1
OS Version:         iOS 10.3.4



NSInternalInconsistencyException: An instance 0x1456e0c0 of class BugsnagConfiguration was deallocated while key value observers were still registered with it. Current observation info: <NSKeyValueObservationInfo 0x1476bc20> (
<NSKeyValueObservance 0x1476cca0: Observer: 0x1474eaf0, Key path: releaseStage, Options: <New: YES, Old: NO, Prior: NO> Context: 0x0, Property: 0x1476cc60>
)

0  CoreFoundation   ___exceptionPreprocess
1  libobjc.A.dylib  _objc_exception_throw
2  CoreFoundation   +[NSException raise:format:]
3  Foundation       _NSKVODeallocate
4  libobjc.A.dylib  objc_object::sidetable_release(bool)
5  ShareMediaEx     -[BugsnagApiClient .cxx_destruct] (BugsnagApiClient.m:19:17)
6  libobjc.A.dylib  object_cxxDestructFromClass(objc_object*, objc_class*)
7  libobjc.A.dylib  _objc_destructInstance
8  libobjc.A.dylib  _object_dispose
9  ShareMediaEx     -[BugsnagApiClient dealloc] (BugsnagApiClient.m:140:1)
10 libobjc.A.dylib  objc_object::sidetable_release(bool)
11 libobjc.A.dylib  object_cxxDestructFromClass(objc_object*, objc_class*)
12 libobjc.A.dylib  _objc_destructInstance
13 libobjc.A.dylib  _object_dispose
14 libobjc.A.dylib  objc_object::sidetable_release(bool)
15 ShareMediaEx     -[BugsnagCrashSentry install:apiClient:onCrash:] (BugsnagCrashSentry.m:25:34)
16 ShareMediaEx     -[BugsnagNotifier start] (BugsnagNotifier.m:326:5)
17 ShareMediaEx     +[Bugsnag startBugsnagWithConfiguration:] (Bugsnag.m:57:13)
18 ShareMediaEx     +[Bugsnag startBugsnagWithApiKey:] (Bugsnag.m:49:5)
19 ShareMediaEx     loadView (ShareMediaController.swift:25:17)
20 ShareMediaEx     loadView (<compiler-generated>)
21 UIKit            -[UIViewController loadViewIfRequired]
22 UIKit            -[UINavigationController _layoutViewController:]
23 UIKit            -[UINavigationController _updateScrollViewFromViewController:toViewController:]
24 UIKit            -[UINavigationController _startTransition:fromViewController:toViewController:]
25 UIKit            -[UINavigationController _startDeferredTransitionIfNeeded:]
26 UIKit            -[UINavigationController __viewWillLayoutSubviews]
27 UIKit            -[UILayoutContainerView layoutSubviews]
28 UIKit            -[UIView(CALayerDelegate) layoutSublayersOfLayer:]
29 QuartzCore       -[CALayer layoutSublayers]
30 QuartzCore       CA::Layer::layout_if_needed(CA::Transaction*)
31 QuartzCore       CA::Layer::layout_and_display_if_needed(CA::Transaction*)
32 QuartzCore       CA::Context::commit_transaction(CA::Transaction*)
33 QuartzCore       CA::Transaction::commit()
34 QuartzCore       CA::Transaction::observer_callback(__CFRunLoopObserver*, unsigned long, void*)
35 CoreFoundation   ___CFRUNLOOP_IS_CALLING_OUT_TO_AN_OBSERVER_CALLBACK_FUNCTION__
36 CoreFoundation   ___CFRunLoopDoObservers
37 CoreFoundation   ___CFRunLoopRun
38 CoreFoundation   _CFRunLoopRunSpecific
39 CoreFoundation   _CFRunLoopRunInMode
40 GraphicsServices _GSEventRunModal
41 UIKit            _UIApplicationMain
42 libxpc.dylib     __xpc_objc_main
43 libxpc.dylib     _xpc_main
44 Foundation       -[NSXPCListener resume]
45 PlugInKit        -[PKService run]
46 PlugInKit        +[PKService main]
47 PlugInKit        +[PKService _defaultRun:arguments:]
48 Foundation       _NSExtensionMain
49 libdyld.dylib    _start
```

**Comment by kattrali** [comment]
Thanks for the report, @rob-keepsafe. Taking a look. :eyes:

**Comment by rob-keepsafe** [comment]
Thanks @kattrali 🙌

**Comment by abigail-bugsnag** [comment]
Released in [v5.22.7](https://github.com/bugsnag/bugsnag-cocoa/releases/tag/v5.22.7).
